[#692] Fix MeshSync test flake & Refactor Client usage

Amr-Shams opened the pull request:
# Description
  Standardizes the Broker and MeshSync packages to use the controller-runtime client interface (client.Client) instead of the raw kubernetes.Clientset. This aligns with the operator
  pattern, leveraging the local Informer cache for health checks and endpoint retrieval to reduce API server load.

  # Fixes
   * Refactor: Updated pkg/broker and pkg/meshsync helper functions to accept and use the controller's cached client.
   * Fix [Bug]: Flaky MeshSync Test #692: Updated meshsync_controller_test.go to wrap update logic in an Eventually block. This resolves race conditions and "stale object" errors during
     high-concurrency test runs caused by the eventual consistency of the local cache.

 # Notes for Reviewers
  This change shifts our health checks to use the local client cache as the first layer (which is updated asynchronously by the Watcher).

   * Regarding the Test Fix: Because we switched to the cache, there is a tiny window of "eventual consistency" where the local view might be slightly behind the API server. In
     high-frequency tests, this caused conflicts.
   * The Solution: I used Eventually in the test to retry the operation. This waits for the cache to catch up with the API server, ensuring the test passes reliably without sacrificing the
     performance benefits of using the cache in production.

 **Signed commits**
  Yes, I signed my commits
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

n2h9 reviewed (commented):
Hello 👋!

Thank you for your contribution :+1: .

1)
Could you please provide an example of failing tests, preferably with a link to ci run (you can add it to the issue).
I have checked the GitHub actions and found couple cases where unit tests are failing, f.e. 
https://github.com/meshery/meshery-operator/actions/runs/20089001041/job/57632612742#step:5:79 
```
[FAILED] Expected an error to have occurred.  Got:
      <nil>: nil
  In [It] at: /home/runner/work/meshery-operator/meshery-operator/controllers/broker_controller_test.go:100 @ 12/10/25 06:11:16.801
```

Is it what this pr fixing?

2) right now lint step is failing, and it prevents code coverage from run, so we can not validate current fix in CI. 
https://github.com/meshery/meshery-operator/issues/694 if by chance you can handle this issue, we could then validate current pr in CI.

Thank you.

n2h9 reviewed (changes requested):
Hello 👋.

1) The DCO check is failing, you need to sign your commits.

2) I didn't ask, to add issue to the commit message, I was asking to get a link to CI run to a test which is failing and which you are fixing. I think I understand that there is no such run, you did some updates and found some flakiness.

That is good, I appreciate your efforts 👍 .
Could you please explain a little bit more: I was doing this, because this, here is the test run output which shows failing test, I fixed it by this. 

3) lint step is still failing, and if you fixing it, please put it in a separate pr, so we can merge lint fix and discuss this test change in this pr.

Thank you 🙇‍♀️.

Amr-Shams commented:
> Hello 👋.
> 
>     1. The DCO check is failing, you need to sign your commits.
> 
>     2. I didn't ask, to add issue to the commit message, I was asking to get a link to CI run to a test which is failing and which you are fixing. I think I understand that there is no such run, you did some updates and found some flakiness.
> 
> 
> That is good, I appreciate your efforts 👍 . Could you please explain a little bit more: I was doing this, because this, here is the test run output which shows failing test, I fixed it by this.
> 
>     3. lint step is still failing, and if you fixing it, please put it in a separate pr, so we can merge lint fix and discuss this test change in this pr.
> 
> 
> Thank you 🙇‍♀️.

oh sorry,i have added the new pull request to update the config file, 
regarding the original change here is this The performance gain is structural: we are replacing a network round-trip (HTTP request to API server) with a local memory lookup (reading from the Informer cache). and hhis standard controller-runtime pattern protects the API server from request flooding during high-concurrency reconciliation. 
then The test update was necessary precisely because we moved to this Eventual Consistency model, requiring us to wait for the watch event to update the local cache before verifying the state.
